nodecg-io-curseforge/extension/curseforgeClient.ts

```diff
@@ -0,0 +1,1034 @@
+     */
+    slug: string;
+    /**
+     * The url to the avatar
+     */
+    avatarUrl: string;
+    /**
+     * The date the category was last updated
+     */
+    dateModified: string;
+    /**
+     * The parent game category id
+     */
+    parentGameCategoryId: number;
+    /**
+     * The root game category id
+     */
+    rootGameCategoryId: number;
+    /**
+     * The game id the category belongs to
+     */
+    gameId: number;
+};
+
+export type CurseDependency = {
+    /**
+     * The addon id
+     */
+    addonId: number;
+    /**
+     * The CurseDependencyType
+     */
+    type: number;
+};
+
+export type CurseModule = {
+    /**
+     * The folder/file name
+     */
+    foldername: string;
+    /**
+     * The folder/file fingerprint
+     */
+    fingerprint: number;
+    /**
+     * The folder/file type
+     */
+    type: number;
+};
+
+export type CurseSortableGameVersion = {
+    gameVersionPadded: string;
+    gameVersion: string;
+    gameVersionReleaseDate: string;
+    gameVersionName: string;
+};
+
+export type CurseFingerprintResponse = {
+    isCacheBuilt: boolean;
+    /**
+     * All exact matches of the given fingerprints
+     */
+    exactMatches: CurseFileInfo[];
+    /**
+     * The fingerprints which matched
+     */
+    exactFingerprints: number[];
+    /**
+     * All files which matched partially
+     */
+    partialMatches: CurseFileInfo[];
+    /**
+     * The fingerprints which matched partially
+     */
+    partialMatchFingerprints: number[];
+    /**
+     * All fingerprints you sent
+     */
+    installedFingerprints: number[];
+    /**
+     * All fingerprints which didn't match
+     */
+    unmatchedFingerprints: number[];
+};
+
+export type CurseFeaturedAddonsResponse = {
+    /**
+     * All featured files which matched the query
+     */
+    Featured: CurseFileInfo[];
+    /**
+     * All popular files which matched the query
+     */
+    Popular: CurseFileInfo[];
+    /**
+     * All recently updated files which matched the query
+     */
+    RecentlyUpdated: CurseFileInfo[];
+};
+
+export type CurseReleaseType = "release" | "beta" | "alpha";
+
+export type CurseDependencyType = "embedded_library" | "optional" | "required" | "tool" | "incompatible" | "include";
+
+export type CurseFileStatus =
+    | "status_1"
+    | "status_2"
+    | "status_3"
+    | "approved"
+    | "rejected"
+    | "status_6"
+    | "deleted"
+    | "archived";
+
+export type CurseProjectStatus =
+    | "new"
+    | "status_2"
+    | "status_3"
+    | "approved"
+    | "status_5"
+    | "status_6"
+    | "status_7"
+    | "status_8"
+    | "deleted";
+
+export type CurseSearchQuery = {
+    categoryID?: number;
+    gameId: number;
+    gameVersion?: string;
+    index?: number;
+    pageSize?: number;
+    searchFilter?: string;
+    sectionId?: number;
+    sort?: number;
+};
+
+export type CurseFeaturedAddonsQuery = {
+    GameId: number;
+    addonsIds?: number[];
```

Comment:
According to the [docs](https://twitchappapi.docs.apiary.io/#/reference/0/get-featured-addons/get-featured-addons/200?mc=reference%2F0%2Fget-featured-addons%2Fget-featured-addons%2F200) this should be `addonIds` (without the s after addons)
```suggestion
    addonIds?: number[];
```